src/main-process/atom-application.coffee

```diff
@@ -648,6 +648,41 @@ class AtomApplication
   #   :devMode - Boolean to control the opened window's dev mode.
   #   :safeMode - Boolean to control the opened window's safe mode.
   openUrl: ({urlToOpen, devMode, safeMode, env}) ->
+    parsedUrl = url.parse(urlToOpen)
+    return unless parsedUrl.protocol is "atom:"
+
+    pack = @findPackageWithName(parsedUrl.host, devMode)
+    if pack?.urlMain
+      @openPackageUrlMain(parsedUrl.host, pack.urlMain, urlToOpen, devMode, safeMode, env)
+    else
+      @openWithAtomUrl(urlToOpen, devMode, safeMode, env)
+
+  openWithAtomUrl: (url, devMode, safeMode, env) ->
```

Comment:
I'm a bit confused about what makes this an "Atom" URL. I wonder if the method could indicate the distinction between the package URL main more clearly?